### DISCUSSION TITLE
Mapping URL's

### DISCUSSION THREAD

**Discussion by thomasverleye** [vercel/next.js#27621] [Help]
Hi Y'all 👋  

We're using Contentful with NextJS to create a new client website.
This website has multiple locales and dynamic routing.

One of the problems we stumble upon is reconstructing our URLs for different locales (to use it inside a language switch, adding them as a Hreflang alternate urls or build a sitemap).

Currently we constructing our urls manually which feels a bit dirty and seems like a foot gun.
The way Link works with an urlObject seems really clean and we were wondering if it's safe to use the [`resolveHref`]( https://github.com/vercel/next.js/blob/0af3b526408bae26d6b3f8cab75c4229998bf7cb/packages/next/next-server/lib/router/router.ts#L261) for this?

Or is there a better way to construct URLs in Next which we're not aware of? 🤔

**Comment by thepuzzlemaster** [comment]
(sorry, I'm not super familiar with the discussions protocol. I'm definitely not suggesting an answer, but rather am experiencing the same issue and would like to share my use case as well. I don't see a way to reply other than "Answer")

I also have this same issue, but for a slightly different reason.

I have an HOC which wraps my a tags to provide consistent visual focus states for all interactive elements. But when I use that HOC inside a &lt;Link&gt;, it doesn't recognize the component as an a tag, even if it ultimately becomes an a tag when the HOC renders its child.

What I'd like to do, is resolve the href which is applied to the Link as a regular href string, and then add that manually to my element so it actually becomes a link.

I came across the resolveHref method which does exactly that, but it's not publicly vended, so I'm hesitant to reach inside the dist/ to use something like that.

Is there some other way to resolve a URL object passed to a Link's href attribute without actually navigating with the router?

Any guidance would be much appreciated.
Thanks.

**Reply by thomasverleye** [reply]
Hello 👋 

No worries about the discussion protocols ;-)

I your problem would be solved if you pass the href in your HOC component.
Check-out this paragraph in the docs:
https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag

Cheers,
Thomas

**Reply by thepuzzlemaster** [reply]
I couldn't get it to work with the `passHref` prop. It's possible that the problem comes from the fact that my HOC, is not officially an HOC, but rather a React function component, which returns a clone of `props.chlidren` with some added event listeners. It does forwardRef onto the child though.

The other possible difference is that the child is also not directly an &lt;a&gt; tag, but a react-bootstrap <Button> element, which renders as an anchor tag because of `as='a'` passed in to it. So it's possible that maybe one of those things is not playing nicely with how the `passHref` prop works. I haven't had a chance to look through the Link code to see what it's actually doing, but I'm sure that might be insightful.
